src/main/java/edu/ucsb/cs156/gauchoride/config/SecurityConfig.java

```diff
@@ -88,6 +88,13 @@ private GrantedAuthoritiesMapper userAuthoritiesMapper() {
           if (email.endsWith("@ucsb.edu")) {
             mappedAuthorities.add(new SimpleGrantedAuthority("ROLE_MEMBER"));
           }
+
+          Optional<User> u = userRepository.findByEmail(email);
+          if (u.isPresent() && u.get().getDriver()) {
+            mappedAuthorities.add(new SimpleGrantedAuthority("ROLE_DRIVER"));
+          } else {
+            mappedAuthorities.add(new SimpleGrantedAuthority("ROLE_RIDER"));
+          }
```

Comment:
It might be good to separate this out into it's own PR so that you can get it merged quickly, and get a separate 10 points for it.

Reply:
Other folks may be depending on this functionality, and you can get this merged quickly; while the changes to the controllers below may take a little longer (since there are several issues that need to be addressed.)

Reply:
Created separate pr